hikari/embeds.py

```diff
@@ -991,3 +991,26 @@ def __eq__(self, other: typing.Any) -> bool:
             else:
                 return True
         return False
+
+    def total_length(self) -> int:
+        """Get the total character count of the embed.
+
+        Returns
+        -------
+        builtins.int
+            The total character count of this embed, including title, description,
+            fields, footer, and author combined.
+        """
+        total = len(self._title or "") + len(self._description or "")
+
+        if self._fields:
+            for field in self._fields:
+                total += len(field.name) + len(field.value)
+
+        if self._footer is not None:
+            total += len(self._footer.text or "")
+
+        if self._author is not None:
+            total += len(self._author.name or "")
```

Comment:
just a random style question but for this and the footer one could this not just be
```py
if self._author and self._author.name:
    total += len(self._author.name)
```
feel like that might be a bit cleaner

Reply:
I had a try at using `getattr` when this got opened. It's much more efficient than checking it twice, but it makes it a bit awkward to read.

Reply:
Changed